test/pkey_name_dup.c

```diff
@@ -0,0 +1,63 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <openssl/bio.h>
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+
+#include "testutil.h"
+
+static const char *algs[] = {
+#ifndef OPENSSL_NO_ECX
+    "ED25519",
+#endif
+#ifndef OPENSSL_NO_ML_KEM
+    "ML-KEM-512",
+#endif
+#ifndef OPENSSL_NO_ML_DSA
+    "ML-DSA-44",
+#endif
+    NULL
+};
+
+static int test_name_dup(void)
+{
+    const char **alg;
+    EVP_PKEY *key = NULL;
+    EVP_PKEY_CTX *factory = NULL, *ctx = NULL;
+    int i, ret = 0;
+
+    for (alg = algs; *alg != NULL; ++alg) {
+        EVP_PKEY_CTX_free(factory);
+
+        if (!TEST_ptr(factory = EVP_PKEY_CTX_new_from_name(NULL, *alg, NULL)))
+            goto end;
+
+        for (i = 0; i < 10; ++i) {
```

Comment:
Isn't doing the dup 10 times an overkill? Yeah, for these particular key types chosen for the test it probably does not take too much time but anyway.

Reply:
Twice should be enough.  Changed accordingly.